test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterHandle.java

```diff
@@ -152,13 +152,34 @@ private WaitForHttpResource configureWaitForReady() throws MalformedURLException
         User credentials = node.getSpec().getUsers().get(0);
         wait.setUsername(credentials.getUsername());
         wait.setPassword(credentials.getPassword());
-        if (securityAutoConfigured) {
+        if (sslEnabled) {
+            configureWaitSecurity(wait, node);
+        } else if (securityAutoConfigured) {
             wait.setCertificateAuthorities(node.getWorkingDir().resolve("config/certs/http_ca.crt").toFile());
         }
 
         return wait;
     }
 
+    private void configureWaitSecurity(WaitForHttpResource wait, Node node) {
+        String caFile = node.getSpec().getSetting("xpack.security.http.ssl.certificate_authorities", null);
+        if (caFile != null) {
+            wait.setCertificateAuthorities(node.getWorkingDir().resolve("config").resolve(caFile).toFile());
+        }
+        String sslCertFile = node.getSpec().getSetting("xpack.security.http.ssl.certificate", null);
+        if (sslCertFile != null) {
+            wait.setCertificateAuthorities(node.getWorkingDir().resolve("config").resolve(sslCertFile).toFile());
+        }
+        String sslKeystoreFile = node.getSpec().getSetting("xpack.security.http.ssl.keystore.path", null);
+        if (sslKeystoreFile != null && caFile == null) { // Can not set both trust stores and CA
+            wait.setTrustStoreFile(node.getWorkingDir().resolve("config").resolve(sslKeystoreFile).toFile());
+        }
+        String keystorePassword = node.getSpec().getSetting("xpack.security.http.ssl.keystore.secure_password", null);
+        if (keystorePassword != null) {
+            wait.setTrustStorePassword(keystorePassword);
+        }
+    }
```

Comment:
This method perpetuates a weirdness, but I have no better immediate suggestion.
Logically speaking, a server doesn't need to know what CA are used to anchor trust to that service. It just knows that it presents a certificate (chain) to clients - it's the client's choice as to which certificate in that chain (or even further up the chain, beyond what is directly presented by the server) should be trusted.

We have this same issue in our existing CLI tools that access the node via HTTP (`elasticsearch-setup-passwords` and `elasticsearch-reset-password`). They need to know which trust anchors to use, but there's nothing in the node's config that is designed to tell them that. They just know which certificates the node will present.

What we do in those tools (today) is similar to what is being done here - we assume that the config that is used to configure the HTTP server can be roughly translated into a config for the HTTP client. It works, some of the time, but not 100%, and it's technically wrong, it just happens to work well enough.

What we'd like to do is switch to a form of certificate pinning. We can inspect the HTTP server config and make a determine which certificate the server will present (or at least, in the case of a keystore, the set of certificates from which the server will pick) and then simply trust any of those certificates.

I don't expect this PR to do that, but I might try and whip up a change to `WaitForHttpResource` that can trust explicit leaf certificates instead of CAs.

Reply:
Why is the server even configured with a CA for HTTP to begin with? I'd assume that is to configure it's own truststore for HTTP communication with other nodes but does ES talk to other nodes over HTTP? I assumed that was all done over the transport protocol?

Reply:
It seems like we probably _could_ simplify this as-is though. (Full disclosure, I stole this code from the existing test clusters stuff).

We first look for a configured CA cert, then look for the actual SSL cert and use that as the CA. I see no instance in which SSL would be enabled and the `http.ssl.certificate` setting would not be configured. Is that correct?

If so we can just remove the bit that looks for the `certificate_authorities` setting altogether.

Reply:
> Why is the server even configured with a CA for HTTP to begin with?

In theory, that should only be used if the node validates client certificates (e.g. PKI realm). As far as ES itself is concerned there is zero relationship between the CA(s) in `x.s.http.ssl.certificate_authorities` and the certificate in `x.s.http.ssl.certificate`. The former is used if we ever need to validate trust to a client cert, and the latter is used as a server cert. It is rare (and coincidental) for them to be the same trust anchors.

In practice, because of this sort of code (in test clusters, and in setup-passwords) sometimes ``x.s.http.ssl.certificate_authorities` is used on a node that doesn't use client certs, as a way of hinting about which CA can be used to anchor to the server's cert (`x.s.http.ssl.certificate`).

> I'd assume that is to configure it's own truststore for HTTP communication with other nodes but does ES talk to other nodes over HTTP? I assumed that was all done over the transport protocol?

We don't currently use HTTP between nodes (though there are some discussions about doing so at some point in the future, perhaps using HTTP/2 as a basis for internode communication).
We do make outgoing HTTP calls for things like watcher (webhooks, sending messages to Slack, etc) and OIDC (communication with authentication servers) however there are separate SSL settings for those (to make things confusing, watcher uses `xpack.http.ssl.*` without the ".security." bit).

> I see no instance in which SSL would be enabled and the http.ssl.certificate setting would not be configured. Is that correct?

In practice we typically us `x.s.http.ssl.certificate` because it uses PEM which works on FIPS so the same tests can be run on both FIPS and non-FIPS CI. But technically, it's possible to use ``x.s.http.ssl.keystore.path` instead of a PEM certificate, and we probably do that in some places (perhaps for tests that we know aren't meaningful on FIPS).

> then look for the actual SSL cert and use that as the CA

From memory that used to work in older JDKs - you could use the builtin truststore mechanic to anchor trust to a specific certificate even if that cert was a leaf certificate and wasn't valid as a CA. But I don't think it works in current JDKs - you can only configure trust against a certificate that has the CA flag set.

> As you say, just trust the configured cert and we're good to go.

I opened https://github.com/elastic/elasticsearch/pull/92657 that would do exactly this. It's a custom trust manager that just checks that the leaf certificate is one of the expected node certificates (in theory, a keystore could contains multiple certs and the runtime cert would be picked based on SNI, or other TLS parameters, so my PR assumes that all of them are trusted and any one of them should be considered valid).